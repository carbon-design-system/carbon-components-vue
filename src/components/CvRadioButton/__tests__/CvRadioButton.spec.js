import { shallowMount } from '@vue/test-utils';
import { CvRadioButton, CvRadioGroup } from '../index';

describe('CvRadioButton', () => {
  // ***************
  // SNAPSHOT CHECKS
  // ***************
  it('matches render with right label', async () => {
    const wrapper = shallowMount(CvRadioButton, {
      props: {
        label: 'right label',
        labelLeft: false,
        id: '1',
        value: '',
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('matches render with left label', async () => {
    const wrapper = shallowMount(CvRadioButton, {
      props: {
        label: 'left label',
        labelLeft: true,
        id: '1',
        value: '',
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('matches render when checked', async () => {
    const wrapper = shallowMount(CvRadioButton, {
      props: {
        label: 'test label',
        labelLeft: false,
        id: '1',
        value: '',
        checked: true,
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('matches render when modelValue is equal to value', async () => {
    const wrapper = shallowMount(CvRadioButton, {
      props: {
        label: 'test label',
        labelLeft: false,
        id: '1',
        value: 'test value',
        modelValue: 'test value',
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('matches render when modelValue is not equal to value', async () => {
    const wrapper = shallowMount(CvRadioButton, {
      props: {
        label: 'test label',
        labelLeft: false,
        id: '1',
        value: 'test value',
        modelValue: 'not test value',
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  // ***************
  // FUNCTIONAL CHECKS
  // ***************
  it('is not checked when checked and modelValue props are not set', async () => {
    const wrapper = shallowMount(CvRadioButton, {
      props: {
        labelLeft: false,
        id: '1',
        value: 'test value',
      },
    });
    expect(wrapper.vm.isChecked).toEqual(false);
  });

  it('is checked when checked prop is set to true and modelValue prop is not set', async () => {
    const wrapper = shallowMount(CvRadioButton, {
      props: {
        labelLeft: false,
        id: '1',
        value: 'test value',
        checked: true,
      },
    });
    expect(wrapper.vm.isChecked).toEqual(true);
  });

  it('modelValue prop has higher priority than checked prop', async () => {
    const wrapper = shallowMount(CvRadioButton, {
      props: {
        labelLeft: false,
        id: '1',
        value: 'test value',
        checked: true,
        modelValue: 'not test value',
      },
    });
    expect(wrapper.vm.isChecked).toEqual(false);
  });
});

describe('CvRadioGroup', () => {
  // ***************
  // SNAPSHOT CHECKS
  // ***************
  it('matches render when vertical', async () => {
    const wrapper = shallowMount(CvRadioGroup, {
      props: { vertical: true },
      slots: {
        default: '<div class="cv-radio-button-stub">RadioButtonStub</div>',
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('matches render when horizontal', async () => {
    const wrapper = shallowMount(CvRadioGroup, {
      props: { vertical: false },
      slots: {
        default: '<div class="cv-radio-button-stub">RadioButtonStub</div>',
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  // ***************
  // FUNCTIONAL CHECKS
  // ***************

  it.only('CvRadioButton renders slot content', async () => {
    const stubId = 'radioBtn';
    const radioButtonStub = `<cvradiobutton id="${stubId}"></cvradiobutton>`;
    const wrapper = shallowMount(CvRadioGroup, {
      slots: {
        default: radioButtonStub,
      },
    });

    const slotNode = wrapper.find('#radioBtn');

    expect(slotNode.html()).toBe(radioButtonStub);
  });
});
