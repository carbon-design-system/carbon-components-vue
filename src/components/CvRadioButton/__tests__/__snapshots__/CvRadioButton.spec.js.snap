// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CvRadioButton matches render when checked 1`] = `<div class="cv-radio-button bx--radio-button-wrapper"><input id="a9fce735-250f-4cf5-b8f7-67d4e6a36515" class="bx--radio-button" type="radio" value=""><!-- symbol causes problem in codepen? --><label for="a9fce735-250f-4cf5-b8f7-67d4e6a36515" class="bx--radio-button__label"><span class="bx--radio-button__appearance"></span><span class="">test label</span></label></div>`;

exports[`CvRadioButton matches render when modelValue is equal to value 1`] = `<div class="cv-radio-button bx--radio-button-wrapper"><input id="462d9c61-1c9f-4344-a8bf-52e2578c85c6" class="bx--radio-button" type="radio" value="test value"><!-- symbol causes problem in codepen? --><label for="462d9c61-1c9f-4344-a8bf-52e2578c85c6" class="bx--radio-button__label"><span class="bx--radio-button__appearance"></span><span class="">test label</span></label></div>`;

exports[`CvRadioButton matches render when modelValue is not equal to value 1`] = `<div class="cv-radio-button bx--radio-button-wrapper"><input id="531901de-675a-4f39-acc9-1a7d0736dd42" class="bx--radio-button" type="radio" value="test value"><!-- symbol causes problem in codepen? --><label for="531901de-675a-4f39-acc9-1a7d0736dd42" class="bx--radio-button__label"><span class="bx--radio-button__appearance"></span><span class="">test label</span></label></div>`;

exports[`CvRadioButton matches render with left label 1`] = `<div class="cv-radio-button bx--radio-button-wrapper bx--radio-button-wrapper--label-left"><input id="6ae49b22-c453-4b3f-85c0-1f294c7eceed" class="bx--radio-button" type="radio" value=""><!-- symbol causes problem in codepen? --><label for="6ae49b22-c453-4b3f-85c0-1f294c7eceed" class="bx--radio-button__label"><span class="bx--radio-button__appearance"></span><span class="">left label</span></label></div>`;

exports[`CvRadioButton matches render with right label 1`] = `<div class="cv-radio-button bx--radio-button-wrapper"><input id="bd6434d7-7761-4fd3-8c78-94fb1383d23c" class="bx--radio-button" type="radio" value=""><!-- symbol causes problem in codepen? --><label for="bd6434d7-7761-4fd3-8c78-94fb1383d23c" class="bx--radio-button__label"><span class="bx--radio-button__appearance"></span><span class="">right label</span></label></div>`;

exports[`CvRadioGroup matches render when horizontal 1`] = `
<div class="cv-radio-group bx--form-item">
  <div class="bx--radio-button-group">
    <div class="cv-radio-button-stub">StubbyMcStubFace</div>
  </div>
</div>
`;

exports[`CvRadioGroup matches render when vertical 1`] = `
<div class="cv-radio-group bx--form-item">
  <div class="bx--radio-button-group bx--radio-button-group--vertical">
    <div class="cv-radio-button-stub">StubbyMcStubFace</div>
  </div>
</div>
`;
